lib/internal/errors.js

```diff
@@ -1714,6 +1714,9 @@ E('ERR_UNSUPPORTED_ESM_URL_SCHEME', (url, supported) => {
   msg += `. Received protocol '${url.protocol}'`;
   return msg;
 }, Error);
+E('ERR_UNSUPPORTED_RESOLVE_REQUEST',
+  'Failed to resolve module specifier "%s": Invalid relative url or base scheme is not hierarchical.',
```

Comment:
I think the message should vary based on the scheme. If it’s a `data:` URL, the message should be something like “`import.meta.resolve` of a bare specifier or relative URL from `data:` URLs is not supported.”